core/iwasm/interpreter/wasm_loader.c

```diff
@@ -1691,8 +1691,15 @@ init_function_local_offsets(WASMFunction *func,
     uint32 i, local_offset = 0;
     uint64 total_size = sizeof(uint16) * ((uint64)param_count + local_count);
 
-    if (!(func->local_offsets =
-                loader_malloc(total_size, error_buf, error_buf_size))) {
+    /*
+     * When total_size is 0, the return value of malloc(0)
+     * is NULL on some platform.
+     * Note: If size is 0, then malloc() returns either NULL,
+     * or a unique pointer value that can later be successfully passed to free().
+     */
```

Comment:
The comments make a little confused, could we change to be like:
Only allocate memory when total_size is not 0, or the return value of malloc(0) might be NULL on some platforms, which causes wasm loader return false.